### ISSUE TITLE
Get NaN on validation metrics

### ISSUE THREAD

**Issue by hzwer** [georgeliu233/STrajNet#2]
When testing the trained model, get NaN on validation set
<img width="779" alt="image" src="https://user-images.githubusercontent.com/10103856/192225020-565691a9-4c98-4b42-996b-8aef0774bfea.png">

**Comment by ehsan-ami** [comment]
I had a similar issue. The last batch of the validation dataset is incomplete and causes the problem in the mirrored strategy. Setting the "drop_remainder=True" for the validation dataset worked for me. 

In train.py:
`valid_dataset = valid_dataset.batch(BATCH_SIZE, drop_remainder=True)`
